robolectric/src/main/java/org/robolectric/internal/bytecode/InstrumentingClassLoader.java

```diff
@@ -380,7 +437,45 @@ public void instrument() {
 
       Set<String> foundMethods = new HashSet<>();
       List<MethodNode> methods = new ArrayList<>(classNode.methods);
+      if (isGraphicsEnabled) {
+
+      }
+      String delegateAnnotationName = "L" + LayoutlibDelegate.class.getName().replace('.', '/') + ";";
       for (MethodNode method : methods) {
+        if (isGraphicsEnabled) {
+          if (method.name.equals("draw")) {
+            continue;
+          }
+          if (method.visibleAnnotations != null) {
+            for (Object visibleAnnotation : method.visibleAnnotations) {
+              AnnotationNode node = (AnnotationNode) visibleAnnotation;
+              if (node.desc.equals(delegateAnnotationName))
+                // System.err.println("Caught IT:" + internalClassName + ":" + method.name);
+                continue;
+            }
+          }
+
+          /*if (internalClassName.replace('/', '.').equals(Bitmap.class.getName())
+           * || internalClassName.replace('/', '.').equals(AccessibilityManager.class.getName())
+           * || internalClassName.replace('/', '.').equals(Paint.class.getName())
+           * || internalClassName.replace('/', '.').equals(TextPaint.class.getName())
+           * || internalClassName.replace('/', '.').equals(BidiRenderer.class.getName())) {*/
+          // System.err.println("Bitmap:" + method.name + "---" + (method.access & ACC_NATIVE));
+          if (internalClassName.replace('/', '.').startsWith("android.graphics.") ||
```

Comment:
But shadowPaintTest is now failing on 21 along as a lot of other tests, because a rendersession is not initiated and resources are not properly set for delegates...